src/Api.Rest/HttpClient/Data/DataServiceRestClient.cs

```diff
@@ -204,6 +204,35 @@ private async Task<DataServiceFeatureMatrix> GetFeatureMatrixInternal( FetchBeha
 			return _FeatureMatrix;
 		}
 
+		private async Task PollOperationStatus( string statusId, CancellationToken cancellationToken = default )
+		{
+			while( !cancellationToken.IsCancellationRequested )
+			{
+				var operationStatus = await _RestClient.Request<OperationStatusDto>( RequestBuilder.CreateGet( $"PendingOperations/{statusId}" ), cancellationToken ).ConfigureAwait( false );
+				if( operationStatus.ExecutionStatus == OperationExecutionStatusDto.Finished )
+					break;
+				if( operationStatus.ExecutionStatus == OperationExecutionStatusDto.Exception )
+					throw new WrappedServerErrorException( operationStatus.Exception );
+			}
+		}
+
+		private async Task DeleteMeasurementsInternal( List<ParameterDefinition> parameter, CancellationToken cancellationToken )
+		{
+			var featureMatrix = await GetFeatureMatrixInternal( FetchBehavior.FetchIfNotCached, cancellationToken ).ConfigureAwait( false );
+
+			if( featureMatrix.SupportsAsyncMeasurementDeletion )
+			{
+				parameter.Add( ParameterDefinition.Create( "runAsync", "true" ) );

```

Comment:
Hier verlässt du dich darauf, dass die Abarbeitung in jedem Fall asynchron ist und ein "Accepted" zurückkommt. Ich würde das etwas flexibler handhaben und auch ein "Ok" korrekt behandeln. Damit ist für die Zukunft offen, dass wenn der Server weiß, dass das Löschen schnell geht, er nicht den Weg über Polling gehen muss.

Reply:
Erledigt. Jetzt gibt es nur noch einen Fehler, wenn Accepted zurückkommt, aber der Location-Header leer ist.